### PR TITLE
fix: Handle FQCN properly with `leading_backslash_in_global_namespace` option enabled

### PR DESCRIPTION
**Pull request by mvorisek**
fix https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7648#discussion_r1438865147

### CODE REVIEW

**Comment by mvorisek** [comment]
@Wirone would you find a moment for review this PR as well - I need it to finish other PRs :)

File: src/Fixer/Import/FullyQualifiedStrictTypesFixer.php [hunk 1]
```diff
@@ -447,27 +453,30 @@ private function shortenClassIfPossible(Tokens $tokens, array $class, array &$us
             $this->registerSymbolForImport('class', $longTypeContent, $uses, $namespaceName);
         }
 
-        if (str_starts_with($longTypeContent, '\\')) {
-            $typeName = substr($longTypeContent, 1);
+        if (str_starts_with($longTypeContent, '\\') || '' === $namespaceName) {
+            $typeName = str_starts_with($longTypeContent, '\\') ? substr($longTypeContent, 1) : $longTypeContent;
```
Comment (Wirone) [inline src/Fixer/Import/FullyQualifiedStrictTypesFixer.php:457]: Wouldn't this be enough?

```suggestion
            $typeName = ltrim($longTypeContent, '\\');
```

**Review by Wirone — COMMENTED** [review]
As far as I understand the changes, the `leading_backslash_in_global_namespace` option was preventing fixer from removing leading `\`, but it was not ensuring it's added when missing? Looks good in general, one minor improvement requested.

File: src/Fixer/Import/FullyQualifiedStrictTypesFixer.php (hunk 1 above)
Comment (mvorisek) [reply src/Fixer/Import/FullyQualifiedStrictTypesFixer.php]: Why not, landed.

**Review by Wirone — APPROVED** [review]

**Pull request merged**